presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -863,7 +884,12 @@ public PlanWithProperties visitSemiJoin(SemiJoinNode node, PreferredProperties p
                     }
                     else {
                         source = withDerivedProperties(
-                                partitionedExchange(idAllocator.getNextId(), REMOTE_STREAMING, source.getNode(), createPartitioning(sourceSymbols), Optional.empty()),
+                                partitionedExchange(
```

Comment:
Per offline discussion, today for SEMI JOIN we don't get any benefits by materializing the exchange (we will get benefit once we support stage level retry). So we can remove it for now :)